depends/toolchain.cmake.in

```diff
@@ -138,27 +140,31 @@ if(NOT WITH_SQLITE AND "@no_sqlite@" STREQUAL "1")
   set(WITH_SQLITE OFF CACHE STRING "Enable SQLite wallet support.")
 endif()
 
-if(NOT WITH_MINIUPNPC AND "@no_upnp@" STREQUAL "1")
-  set(WITH_MINIUPNPC OFF CACHE STRING "Enable UPnP.")
+if("@no_upnp@")
+  set(WITH_MINIUPNPC OFF CACHE BOOL "")
+else()
+  set(WITH_MINIUPNPC ON CACHE BOOL "")
 endif()
 
-if(NOT WITH_NATPMP AND "@no_natpmp@" STREQUAL "1")
-  set(WITH_NATPMP OFF CACHE STRING "Enable NAT-PMP.")
+if("@no_natpmp@")
+  set(WITH_NATPMP OFF CACHE BOOL "")
+else()
+  set(WITH_NATPMP ON CACHE BOOL "")
 endif()
 
 if(NOT WITH_USDT AND "@no_usdt@" STREQUAL "1")
   set(WITH_USDT OFF CACHE STRING "Enable tracepoints for Userspace, Statically Defined Tracing.")
 endif()
 
-if(NOT HARDENING AND "@no_harden@" STREQUAL "1")
-  set(HARDENING OFF CACHE STRING "Attempt to harden the resulting executables.")
+if("@no_harden@")
+  set(HARDENING OFF CACHE BOOL "")
+else()
+  set(HARDENING ON CACHE BOOL "")
 endif()
 
 if("@multiprocess@" STREQUAL "1")
-  if(NOT MULTIPROCESS)
-    set(MULTIPROCESS ON CACHE STRING "")
-  endif()
-  if(NOT LibmultiprocessGen_DIR)
-    set(LibmultiprocessGen_DIR "${CMAKE_FIND_ROOT_PATH}/native/lib/cmake/LibmultiprocessGen" CACHE PATH "")
-  endif()
+  set(MULTIPROCESS ON CACHE BOOL "")
+  set(LibmultiprocessGen_DIR "${CMAKE_FIND_ROOT_PATH}/native/lib/cmake/LibmultiprocessGen" CACHE PATH "")
+else()
+  set(MULTIPROCESS OFF CACHE BOOL "")
 endif()
```

Comment:
`MULTIPROCESS` is a boolean like e.g. `WITH_NATPMP` but is handled differently here. Why? Should this not be:

```cmake
if("@no_multiprocess@")
  set(MULTIPROCESS OFF CACHE BOOL "")
else()
  set(MULTIPROCESS ON CACHE BOOL "")
  set(LibmultiprocessGen_DIR "${CMAKE_FIND_ROOT_PATH}/native/lib/cmake/LibmultiprocessGen" CACHE PATH "")
endif()
```
?